### DISCUSSION TITLE
Post v1.0.0 bug fixes

### DISCUSSION THREAD

**Discussion by e-t-k** [CBIIT/ccdi-federation-api#109] [API Design Discussion]
This discussion is to track bugs, omissions, and errors in the **interface design** or **reference implementation (Rust)** that are still present in the 1.0.0 release and should be resolved.
This is not for new features (see #110 for that) and it is not for problems with any federation member's implementation of the interface.

#### Diagnosis not available as search parameter for sample endpoint
**Bug type**  Omission in interface design
**Endpoint** GET /sample?diagnosis=VALUE ; http://localhost:8000/swagger-ui/#/Sample/ 
**Expected behavior** "diagnosis" field is available for filtering on the swagger page
**Current behavior per API spec** No diagnosis field present
**Is fix a breaking change?** No: a new parameter option will be added which clients may choose to ignore.
**Will implementations need to update?** Yes: if an implementation supports the sample endpoint and *diagnosis* is not yet a searchable parameter it will need to be added.
**Date added** June 24 2024
**Current status** Still present in `main` branch as of Feb 19 2025
**Notes**

#### Diagnosis filtering not implemented in reference implementation
**Bug type**  Error in reference implementation
**Endpoint** GET /sample?diagnosis=VALUE . Example: http://localhost:8000/sample?diagnosis=WrongDiagnosis
**Expected behavior** Samples are filtered by the value of the "diagnosis" field
**Current behavior per API spec** Filter is ignored and all samples are returned
**Is fix a breaking change?** No - reference implementation only
**Will implementations need to update?** No.
**Date added** Feb 19 2025
**Current status** Still present in `main` branch as of Feb 19 2025
**Notes**

#### FIXED Race `Black or African American` has the `Asian` description
~**Bug type** Error in reference implementation
**Endpoint** The wiki: [ccdi-federation-api](https://github.com/CBIIT/ccdi-federation-api/wiki/Subject-Metadata-Fields) / [ccdi-federation-api-spec](https://github.com/CBIIT/ccdi-federation-api-spec/wiki/Subject-Metadata-Fields)
**Expected behavior** wiki has the correct description for  `Black or African American`
**Current behavior** wiki has the wrong description
**Is fix a breaking change?** No: fix applies to wiki only; no client needs to make a change.
**Will implementations need to update?** No;  harmonized fields  in `metadata/fields/subject` do not have a Description attribute.
**Date added** July 3 2024
**Current status** [Fixed on Dec 19 2024](https://github.com/CBIIT/ccdi-federation-api/commit/c8bb066bdf9fe515834dbb2c6d8b28b4e757ce2f)~

**Comment by Ssandor13** [comment]
**Tissue Type has a typo for Begin Date and odd description for Tumor**
Bug type Error in reference implementation
**Endpoint** The wiki: [[[ccdi-federation-api](https://github.com/CBIIT/ccdi-federation-api/wiki/Subject-Metadata-Fields) / [ccdi-federation-api-spec](https://github.com/CBIIT/ccdi-federation-api-spec/wiki/Subject-Metadata-Fields](https://github.com/stjudecloud/ccdi/wiki/Sample-Metadata-Fields#tissue_type))](https://github.com/stjudecloud/ccdi/wiki/Sample-Metadata-Fields#tissue_type)](https://github.com/CBIIT/ccdi-federation-api/wiki/Sample-Metadata-Fields#tissue_type)
**Expected behavior** 8/25/2016 should be formatted correctly
**Current behavior** Begin Date is not formatted correctly nor description
**Is fix a breaking change?** No: fix applies to wiki only; no client needs to make a change.
**Will implementations need to update?** No
**Date added** July 8th, 2024
**Current status** Fixed 2/13/25

**Reply by Ssandor13** [reply]
Manually fixed in the wiki 02/13/25.

**Comment by e-t-k** [comment]
#### organization endpoint calls parameter 'name' but expects 'identifier'
**Bug type**  misleading documentation in swagger.yml leading to inconsistent implementations
**Endpoint** GET /organization/<name> ; http://localhost:8000/organization/example-organization
**Problem** swagger spec suggests that individual `organization` info should be retrieved using the `name` field (which can have special characters); however, reference implementation actually accepts `identifier` field.  This inconsistency caused UCSC,  StJude, PCDC to implement matching identifier and CHOP to implement matching name.
**Resolution** We will standardize on matching `identifier` and update the text in the swagger spec ; CHOP plans to update their implementation.
**Is fix a breaking change?** Yes - clients may have stored queries for CHOP matching old implementation
**Will implementations need to update?**  CHOP only.
**Date added** July 17 2024
**Current status** Still present in `main` branch as of Feb 19 2025
**Notes**
Code to fix: https://github.com/CBIIT/ccdi-federation-api/blob/main/crates/ccdi-server/src/routes/organization.rs#L91
```
    path = "/organization/{name}",
    params(
        (
            "name" = String,
            description = "The name of the organization.",
        ),
```

**Comment by asafieva** [comment]
**Bug type**: index.html "title" and "description" HTML tags were not updated after changes in swagger.yml "info">>"title" and "info">>"description". This makes URL bookmarks appearance inconsistent with swagger website cbiit.github.io/ccdi-federation-api.
**Endpoint**: NA
**Problem**:
title YAML: "CCDI Data Federation: Participating Nodes API"
title index.html: "CCDI Data Federation API"
description in YAML: "The Childhood Cancer Data Initiative (CCDI) Data Federation API supports the querying of federated pediatric cancer within the broader community."
description in index.html: "An API for querying federated pediatric cancer data from the broader community.""
**Resolution**: Update index.html tags to match swagger.yml info title and description elements.
**Is fix a breaking change?** No
**Will implementations need to update?** No.
**Date added:** July 23 2024
**Current status:** fix not yet started as of July 23 2024

**Reply by asafieva** [reply]
index.html "title" and "description" HTML tags were fixed as requested starting v1.0.0.

**Comment by asafieva** [comment]
**Bug type**: sample attributes "age_at_diagnosis" (3225640), "age_at_collection" (14473376), subject attribute "age_at_vital_status" (14480965) do not have CDEs references on wiki.
**Endpoint**: sample, subject
**Problem**: CDE missing references on API wiki pages.
**Resolution**: Add CDE missing references to harmonized values, on API wiki pages.
**Is fix a breaking change**? No
**Will implementations need to update**? Yes.
**Date added**: July 31 2024
**Current status**: fix not yet started as of July 31 2024

**Reply by e-t-k** [reply]
My understanding is this is a deliberate choice to avoid CDEs for these - see eg https://github.com/CBIIT/ccdi-federation-api/pull/154#issuecomment-2884815221 .

**Comment by asafieva** [comment]
**Bug type**: An attribute "_**depositions**_" is not defined as nullable. API group decided on a meeting 09/27/2024 that this is a bug, and "depositions" attributes shall be allowed to be nullable.
**Endpoint**: All using "models.metadata.common.Metadata" component.
**Problem**: An attribute "depositions" is not defined as nullable in the API spec.  But "null" values are sent in API responses besides KidsFirst server. 
**Resolution**: "depositions" attributes shall be defined as nullable in the spec to allow returning the entities which are not related to dbGaP. The requirement to have not-null depositions would force API nodes to exclude a lot of entities to comply with this requirement.
**Is fix a breaking change?** No
**Will implementations need to update?** Yes, "swagger.yml" specification shall be updated.
**Date added:** Sep 27 2024
**Current status:** fix not yet started as of Sep 27 2024
